### PR TITLE
[EN-74] Show salary in Contract view

### PR DESCRIPTION
**Pull request by baremarco**
# Description :pencil:

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.
- Created CustomizableChipField component to render custom content using a function passed as a child.
- Added HasPermission to control visibility of the content (Salary information)
## Link to ticket :link:

https://entropyteam.atlassian.net/browse/EN-74

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? :microscope:

- Was created different types of _Payment settlement_ between the different currency types
-  Was created different types of _Payment settlement_ between the different modality types
- Was created different types of _Payment settlement_ up to three to test the component's arrange for the available space

### CODE REVIEW

File: src/contracts.js [hunk 1]
```diff
@@ -118,6 +124,23 @@ export const ContractList = () => (
         label="Status"
         render={(record) => (record.active === true ? "Active" : "Inactive")}
       />
+      {HasPermissions("contracts", "read") && (
```
Comment (agustinb4) [inline src/contracts.js:127]: You could remove this check, we are checking the permissions at resource level

File: src/contracts.js [hunk 2]
```diff
@@ -118,6 +124,23 @@ export const ContractList = () => (
...
+        <ArrayField source="paymentSettlement" label="Salary">
+          <SingleFieldList>
+            <CustomizableChipField source="salary">
+              {(record) => {
+                if (record) {
+                  const label = `${record.currency}${
+                    record.salary
+                  }/${record.modality.charAt(0).toLowerCase()}`;
+                  return label;
```
Comment (agustinb4) [inline src/contracts.js:136]: For Developer and Analyst role (they cannot see salary), i saw the value of ${record.salary} is null, so i think we could improve this with something like:
```
if (record) {
  const salary = `${record.salary}` === "null" ? '-' : `${record.salary}`
  return `${record.currency} ${salary} /${record.modality.charAt(0).toLowerCase()}`;
}
```

**Review by agustinb4 — CHANGES_REQUESTED** [review]

**Review by agustinb4 — APPROVED** [review]
Looks good!

**Review by FacuOrellana — APPROVED** [review]
nice

**Pull request merged**